sidecar/config/proxy_config.go

```diff
@@ -0,0 +1,88 @@
+// Copyright 2016 IBM Corporation
+//
+//   Licensed under the Apache License, Version 2.0 (the "License");
+//   you may not use this file except in compliance with the License.
+//   You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+//   Unless required by applicable law or agreed to in writing, software
+//   distributed under the License is distributed on an "AS IS" BASIS,
+//   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//   See the License for the specific language governing permissions and
+//   limitations under the License.
+
+package config
+
+import (
+	"os"
+	"strconv"
+	"text/template"
+
+	"github.com/Sirupsen/logrus"
+)
+
+type a8ServicesNginxConfData struct {
+	SSLFlag     string
+	CertPath    string
+	CertKeyPath string
+	CACertPath  string
+}
+
+// Proxy nginx configuration values.
+const (
+	A8ServicesConfTemplatePath        = "/etc/nginx/amalgam8-services.conf.tmpl"
+	A8ServicesConfPath                = "/etc/nginx/amalgam8-services.conf"
+	NginxSSLFlag                      = " ssl"
+	NginxSSLCertDirective             = "ssl_certificate "
+	NginxSSLCertKeyDirective          = "ssl_certificate_key "
+	NginxProxySSLTrustedCertDirective = "proxy_ssl_trusted_certificate "
+)
+
+// GenProxyConfig generates proxy config
+func (c *Config) GenProxyConfig() error {
```

Comment:
> Could we start with this and then you guys decide the best place/structure for it in the context of the envoy changes? And you refactor as appropriate?

My comment above suggests the appropriate place to put the functions/structure. If you want this PR to be merged, please change it.